int32 pin_thumb=0
int32 pin_index=1
int32 pin_middle=2
int32 pin_ring=3
int32 pin_pinky=4
int32 pin_palm=5
int32 n_pins=6
int32 n_levels=5

uint16[6] levels_per_pin